### DISCUSSION TITLE
Can you make Tenants a Requirement?

### DISCUSSION THREAD

**Discussion by eduncan911** [netbox-community/netbox#5956] [Q&A] [answered]
I'm using Tenants to filter multiple customer setups.  But, I am forgetting to always set the Tenant on the Add New screens.  Is there a way to make this a requirement when filling out all Add New forms?

Also, why do I always have to populate this?  When viewing an object (e.g. a Tenant), I can click on "Sites" or "Racks" on the right side - within this Tenant view.  However, the Add New pages do not pre-populate the Tenant - even though I am on the tenant slug?

Thanks!

**Comment by jpobeda** [comment]
I use reports to flag those sort of things. If you forget to add tenants frequently, it's very likely there are other devices without tenant associated and reports can help you to find them quickly and go over them.

You could create a script that does that and forces that field to be required but then you wouldn't be using NB core UI but invoking the script you created. Not sure that'd work for you.

**Reply by eduncan911** [reply]
Interesting, using Reports.  Haven't looked into reporting yet so I'll give that a shot.

**Comment by proudbro** [comment, marked as answer]
You can add new permission for example for devices (dcim.add/change/delete_device) with constraint such as: 
`{"tenant__id__isnull": false}`
And apply it to users or/and groups of users

Now you cannot create/edit/delete devices without the mandatory filling tenant field

**Reply by eduncan911** [reply]
Ah, found where to place constraints: https://netbox.readthedocs.io/en/stable/administration/permissions/

**Reply by PieterL75** [reply]
that's a pretty good approach.. Too bad that the errormessage is not very descriptive
"Errors: Object save failed due to object-level permissions violation"

**Comment by sdktr** [comment]
You can use the CustomValidators capability as well.

**Reply by jeremystretch** [reply]
^ For clarification, custom validation is being introduced in the upcoming v3.0 release. It's not available in the current release.
